mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -43,6 +55,379 @@ struct CompactBlockRow {
+                    Timer::sleep(30.).await;
+                },
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl ZRpcOps for NativeClient {
+    async fn get_block_height(&mut self) -> Result<u64, MmError<UpdateBlocksCacheErr>> {
+        Ok(self.get_block_count().compat().await?)
+    }
+
+    async fn scan_blocks(
+        &mut self,
+        start_block: u64,
+        last_block: u64,
+        on_block: &mut (dyn FnMut(TonicCompactBlock) -> Result<(), MmError<UpdateBlocksCacheErr>> + Send + Sync),
+    ) -> Result<(), MmError<UpdateBlocksCacheErr>> {
+        for height in start_block..=last_block {
+            let block = self.get_block_by_height(height).await?;
+            debug!("Got block {:?}", block);
+            let mut compact_txs = Vec::new();
+            // By default, CompactBlocks only contain CompactTxs for transactions that contain Sapling spends or outputs.
+            // Create and push compact_tx during iteration.
+            for (tx_id, hash_tx) in block.tx.iter().enumerate() {
+                let tx_bytes = self.get_transaction_bytes(hash_tx).compat().await?;
+                let tx = ZTransaction::read(tx_bytes.as_slice()).unwrap();
+                let mut spends = Vec::new();
+                let mut outputs = Vec::new();
+                if !tx.shielded_spends.is_empty() || !tx.shielded_outputs.is_empty() {
+                    // Create and push spends with outs for compact_tx during iterations.
+                    for spend in &tx.shielded_spends {
+                        let compact_spend = TonicCompactSpend {
+                            nf: spend.nullifier.to_vec(),
+                        };
+                        spends.push(compact_spend);
+                    }
+                    for out in &tx.shielded_outputs {
+                        let compact_out = TonicCompactOutput {
+                            cmu: out.cmu.to_bytes().to_vec(),
+                            epk: out.ephemeral_key.to_bytes().to_vec(),
+                            // https://zips.z.cash/zip-0307#output-compression
+                            // The first 52 bytes of the ciphertext contain the contents and opening of the note commitment,
+                            // which is all of the data needed to spend the note and to verify that the note is spendable.
+                            ciphertext: out.enc_ciphertext[0..52].to_vec(),
+                        };
+                        outputs.push(compact_out);
+                    }
+                    // Shadowing mut variables as immutable. No longer need to update them.
+                    drop_mutability!(spends);
+                    drop_mutability!(outputs);
+                    let mut hash_tx_vec = hash_tx.0.to_vec();
+                    hash_tx_vec.reverse();
+
+                    let compact_tx = TonicCompactTx {
+                        index: tx_id as u64,
+                        hash: hash_tx_vec,
+                        fee: 0,
```

Comment:
There is note for fee in CompactTx `The transaction fee: present if server can provide. In the case of a stateless server and a transaction with transparent inputs, this will be unset because the calculation requires reference to prior transactions.`

The light mode all the time puts `fee: 0` for compact transactions.
So, yes, it will be 0.